lms/djangoapps/courseware/tests/test_entrance_exam.py

```diff
@@ -519,3 +569,116 @@ def test_skip_entrance_exame_gating(self):
         )
         for toc_section in self.expected_unlocked_toc:
             self.assertIn(toc_section, unlocked_toc)
+
+    def test_entrance_exam_gating_for_staff(self):
+        """
+        Tests gating is disabled if user is member of staff.
+        """
+
+        # Login as member of staff
+        self.client.logout()
+        staff_user = StaffFactory(course_key=self.course.id)
+        staff_user.is_staff = True
+        self.client.login(username=staff_user.username, password='test')
+
+        # assert staff has access to all toc
+        self.request.user = staff_user
+        unlocked_toc = toc_for_course(
+            self.request,
+            self.course,
+            self.entrance_exam.url_name,
+            self.exam_1.url_name,
+            self.field_data_cache
+        )
+        for toc_section in self.expected_unlocked_toc:
+            self.assertIn(toc_section, unlocked_toc)
+
+    @patch.dict("django.conf.settings.FEATURES", {'ENTRANCE_EXAMS': True})
+    @patch('courseware.entrance_exams.user_has_passed_entrance_exam', Mock(return_value=False))
+    def test_courseware_page_access_without_passing_entrance_exam(self):
+        """
+        Test courseware access page without passing entrance exam
+        """
+        url = reverse(
+            'courseware_chapter',
+            kwargs={'course_id': unicode(self.course.id), 'chapter': self.chapter.url_name}
+        )
+        response = self.client.get(url)
+        redirect_url = reverse('courseware', args=[unicode(self.course.id)])
+        self.assertRedirects(response, redirect_url, status_code=302, target_status_code=302)
+        response = self.client.get(redirect_url)
+        exam_url = response.get('Location')
+        self.assertRedirects(response, exam_url)
+
+    @patch.dict("django.conf.settings.FEATURES", {'ENTRANCE_EXAMS': True})
+    @patch('courseware.entrance_exams.user_has_passed_entrance_exam', Mock(return_value=True))
+    def test_courseware_page_access_after_passing_entrance_exam(self):
+        """
+        Test courseware access page after passing entrance exam
+        """
+        # Mocking get_required_content with empty list to assume user has passed entrance exam
+        url = reverse(
+            'courseware_chapter',
+            kwargs={'course_id': unicode(self.course.id), 'chapter': self.chapter.url_name}
+        )
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    @patch.dict("django.conf.settings.FEATURES", {'ENTRANCE_EXAMS': True})
+    @patch('util.milestones_helpers.get_required_content', Mock(return_value=['a value']))
+    def test_courseware_page_access_with_staff_user_without_passing_entrance_exam(self):
```

Comment:
Must: Though it may be redundant, I'd like to see the test case for a staff user who has also passed an entrance exam.